### ISSUE TITLE
Error while fixing Array<Dictionary<String, Any>>

### ISSUE THREAD

**Issue by federik21** [realm/SwiftLint#3678]
### New Issue Checklist

- [👌] Updated SwiftLint to the latest version
- [👌 ] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Launched swiftlint --fix on my project, Array<Dictionary<String, Any>> was fixed with [Dictionary[String, Any]>

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint --fix
...
/Users/piccirillif/Develop/mybike_v2_app-iOS/MyBike_v2_App/AppDelegate.swift:109:36 Corrected Syntactic Sugar
...
```

### Environment

* SwiftLint 0.43.1
* Installation method Homebrew
* Paste your configuration file:

```swift
// This triggers a violation:
            let array = result as? [Dictionary[String, Any]>
```

**Comment by SimplyDanny** [comment]
This should be fixed with the most recent release [0.47.1](https://github.com/realm/SwiftLint/releases/tag/0.47.1). Could you please confirm, @federik21?

**Comment by jpsim** [comment]
Will close assuming fixed, and can reopen if needed.
